snuba/query/parser/validation/functions.py

```diff
@@ -59,32 +60,30 @@ class QueryEntityFinder(
     a single QueryEntity.
     """
 
-    def _visit_simple_source(self, data_source: QueryEntity) -> QueryEntity:
-        return data_source
+    def _visit_simple_source(self, data_source: QueryEntity) -> List[QueryEntity]:
+        return [data_source]
```

Comment:
Maybe this should return a Set? It's possible in a Join for the same entity to appear multiple times.